--- a/src/io-v4l.c
+++ b/src/io-v4l.c
@@ -45,6 +45,9 @@
 #include <sys/stat.h>		/* S_ISCHR */
 #include <sys/ioctl.h>		/* for (_)videodev.h */
 #include <pthread.h>
+#ifdef HAVE_ANDROID
+#include <signal.h>
+#endif
 
 #include "videodev.h"
 #include "_videodev.h"
@@ -208,7 +211,13 @@
 	v->read_active = TRUE;
 
 	for (;;) {
+#ifndef HAVE_ANDROID
 		pthread_testcancel();
+#else
+		sigset_t waiting_mask;
+		if(sigpending (&waiting_mask))
+			pthread_exit(0);
+#endif
 
 		r = read(v->fd, raw->data, raw->size);
 
--- a/src/io-v4l2k.c
+++ b/src/io-v4l2k.c
@@ -54,6 +54,9 @@
 #include <sys/mman.h>		/* PROT_READ, MAP_SHARED */
 #include <asm/types.h>		/* __u8 and friends for videodev2k.h */
 #include <pthread.h>
+#ifdef HAVE_ANDROID
+#include <signal.h>
+#endif
 
 #include "raw_decoder.h"
 #include "version.h"
@@ -654,7 +657,13 @@
 
 	for (;;) {
 		/* from zapping/libvbi/v4lx.c */
+#ifndef HAVE_ANDROID
 		pthread_testcancel();
+#else
+		sigset_t waiting_mask;
+		if(sigpending (&waiting_mask))
+			pthread_exit(0);
+#endif
 
 		r = read(v->fd, raw->data, raw->size);
 
