diff --git a/xmlIO.c b/xmlIO.c
index 1a79c09..9d8bd15 100644
--- a/xmlIO.c
+++ b/xmlIO.c
@@ -77,6 +77,10 @@
 #include "buf.h"
 #include "enc.h"
 
+#if defined(WINAPI_FAMILY) && (WINAPI_FAMILY == WINAPI_FAMILY_APP)
+#include <io.h>
+#endif
+
 /* #define VERBOSE_FAILURE */
 /* #define DEBUG_EXTERNAL_ENTITIES */
 /* #define DEBUG_INPUT */
@@ -3763,7 +3767,7 @@ xmlParserGetDirectory(const char *filename) {
 	else *cur = 0;
 	ret = xmlMemStrdup(dir);
     } else {
-        if (getcwd(dir, 1024) != NULL) {
+        if (_getcwd(dir, 1024) != NULL) {
	    dir[1023] = 0;
	    ret = xmlMemStrdup(dir);
	}
diff --git a/xmlmemory.c b/xmlmemory.c
index f08c8c3..8b9c35e 100644
--- a/xmlmemory.c
+++ b/xmlmemory.c
@@ -56,6 +56,10 @@ static xmlMutexPtr xmlMemMutex = NULL;
 
 void xmlMallocBreakpoint(void);
 
+#if defined(WINAPI_FAMILY) && WINAPI_FAMILY == WINAPI_FAMILY_APP
+# define getenv(x) NULL
+#endif
+
 /************************************************************************
  *									*
  *		Macros, variables and associated types			*
-- 
2.13.2.windows.1

