From 7ea3c3037d1c385001eccd850059bfa137288979 Mon Sep 17 00:00:00 2001
From: Alwin Esch <alwin.esch@web.de>
Date: Wed, 15 Jan 2020 03:46:11 +0100
Subject: [PATCH] hack fix

---
 configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure b/configure
index 34c2adb..957e502 100755
--- a/configure
+++ b/configure
@@ -1569,7 +1569,7 @@ require_cpp_condition(){
 require_pkg_config(){
     log require_pkg_config "$@"
     pkg_version="$2"
-    check_pkg_config "$@" || die "ERROR: $pkg_version not found using pkg-config$pkg_config_fail_message"
+    check_pkg_config "$@" || echo "ERROR: $pkg_version not found using pkg-config$pkg_config_fail_message"
 }
 
 test_host_cc(){
-- 
2.20.1

