diff -ur FFmpeg-n4.3/configure FFmpeg-n4.3-01/configure
--- FFmpeg-n4.3/configure	2020-06-15 19:41:15.000000000 +0100
+++ FFmpeg-n4.3-01/configure	2020-06-18 09:00:30.000000000 +0100
@@ -6468,6 +6468,8 @@
                                check_lib openssl openssl/ssl.h SSL_library_init -lssl -lcrypto ||
                                check_lib openssl openssl/ssl.h SSL_library_init -lssl32 -leay32 ||
                                check_lib openssl openssl/ssl.h SSL_library_init -lssl -lcrypto -lws2_32 -lgdi32 ||
+                               check_lib openssl openssl/ssl.h OPENSSL_init_ssl -llibssl -llibcrypto -lz -lws2_32 -lgdi32 -ladvapi32 -luser32 ||
+                               check_lib openssl openssl/ssl.h SSL_library_init -llibeay32 -lssleay32 ||
                                die "ERROR: openssl not found"; }
 enabled pocketsphinx      && require_pkg_config pocketsphinx pocketsphinx pocketsphinx/pocketsphinx.h ps_init
 enabled rkmpp             && { require_pkg_config rkmpp rockchip_mpp  rockchip/rk_mpi.h mpp_create &&
