From e70f8aad605e17795d9bc351850ca0a37ad0071d Mon Sep 17 00:00:00 2001
From: Alwin Esch <alwin.esch@web.de>
Date: Wed, 15 Jan 2020 10:04:53 +0100
Subject: [PATCH] fix build test with this as static on ffmpeg

---
 configure | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/configure b/configure
index 957e502..f108564 100755
--- a/configure
+++ b/configure
@@ -7,6 +7,8 @@
 # Copyright (c) 2005-2008 Mans Rullgard
 #
 
+export PKG_CONFIG_PATH="$PKG_CONFIG_PATH:$(pkg-config --variable pc_path pkg-config)"
+
 # Prevent locale nonsense from breaking basic text processing.
 LC_ALL=C
 export LC_ALL
-- 
2.20.1

