diff --git a/libavcodec/aarch64/Makefile b/libavcodec/aarch64/Makefile
index 9ce21566..efe2ce9f 100644
--- a/libavcodec/aarch64/Makefile
+++ b/libavcodec/aarch64/Makefile
@@ -51,7 +51,7 @@ NEON-OBJS-$(CONFIG_MDCT)                += aarch64/mdct_neon.o
 NEON-OBJS-$(CONFIG_ME_CMP)              += aarch64/me_cmp_neon.o
 NEON-OBJS-$(CONFIG_MPEGAUDIODSP)        += aarch64/mpegaudiodsp_neon.o
 NEON-OBJS-$(CONFIG_PIXBLOCKDSP)         += aarch64/pixblockdsp_neon.o
-NEON-OBJS-$(CONFIG_VC1DSP)              += aarch64/vc1dsp_neon.o
+#NEON-OBJS-$(CONFIG_VC1DSP)              += aarch64/vc1dsp_neon.o
 NEON-OBJS-$(CONFIG_VP8DSP)              += aarch64/vp8dsp_neon.o
 
 # decoders/encoders
diff --git a/libavcodec/aarch64/vc1dsp_init_aarch64.c b/libavcodec/aarch64/vc1dsp_init_aarch64.c
index a7976fd5..ef31019b 100644
--- a/libavcodec/aarch64/vc1dsp_init_aarch64.c
+++ b/libavcodec/aarch64/vc1dsp_init_aarch64.c
@@ -26,6 +26,7 @@
 
 #include "config.h"
 
+#if 0
 void ff_vc1_inv_trans_8x8_neon(int16_t *block);
 void ff_vc1_inv_trans_8x4_neon(uint8_t *dest, ptrdiff_t stride, int16_t *block);
 void ff_vc1_inv_trans_4x8_neon(uint8_t *dest, ptrdiff_t stride, int16_t *block);
@@ -109,9 +110,11 @@ static int vc1_unescape_buffer_neon(const uint8_t *src, int size, uint8_t *dst)
     }
     return dsize;
 }
+#endif // #if 0
 
 av_cold void ff_vc1dsp_init_aarch64(VC1DSPContext *dsp)
 {
+#if 0
     int cpu_flags = av_get_cpu_flags();
 
     if (have_neon(cpu_flags)) {
@@ -138,4 +141,5 @@ av_cold void ff_vc1dsp_init_aarch64(VC1DSPContext *dsp)
 
         dsp->vc1_unescape_buffer = vc1_unescape_buffer_neon;
     }
+#endif // #if 0
 }
