diff -ur ffmpeg/libavformat/tls_openssl.c ffmpeg-patch/libavformat/tls_openssl.c
--- ffmpeg/libavformat/tls_openssl.c	2020-01-31 12:12:04.000000000 +0000
+++ ffmpeg-patch/libavformat/tls_openssl.c	2020-01-31 12:13:10.000000000 +0000
@@ -48,7 +48,7 @@
 #endif
 } TLSContext;
 
-#if HAVE_THREADS
+#if HAVE_THREADS && OPENSSL_VERSION_NUMBER < 0x10100000L
 #include <openssl/crypto.h>
 pthread_mutex_t *openssl_mutexes;
 static void openssl_lock(int mode, int type, const char *file, int line)
@@ -79,7 +79,7 @@
         SSL_library_init();
         SSL_load_error_strings();
 #endif
-#if HAVE_THREADS
+#if HAVE_THREADS && OPENSSL_VERSION_NUMBER < 0x10100000L
         if (!CRYPTO_get_locking_callback()) {
             int i;
             openssl_mutexes = av_malloc_array(sizeof(pthread_mutex_t), CRYPTO_num_locks());
@@ -108,7 +108,7 @@
     ff_lock_avformat();
     openssl_init--;
     if (!openssl_init) {
-#if HAVE_THREADS
+#if HAVE_THREADS && OPENSSL_VERSION_NUMBER < 0x10100000L
         if (CRYPTO_get_locking_callback() == openssl_lock) {
             int i;
             CRYPTO_set_locking_callback(NULL);
