--- a/librtmp/hashswf.c
+++ b/librtmp/hashswf.c
@@ -470,6 +470,13 @@
   hpre.av_val = "Q:";
   hpre.av_len = 2;
   home.av_val = "\\UserData";
+#elif defined(WINAPI_FAMILY) && (WINAPI_FAMILY == WINAPI_FAMILY_APP)
+  extern const char *get_home_drive(void);
+  extern char *get_home_path(void);
+
+  hpre.av_val = get_home_drive();
+  hpre.av_len = strlen(hpre.av_val);
+  home.av_val = get_home_path();
 #else
   hpre.av_val = getenv("HOMEDRIVE");
   hpre.av_len = strlen(hpre.av_val);
--- a/librtmp/rtmp.c
+++ b/librtmp/rtmp.c
@@ -158,7 +158,7 @@
 #ifdef _DEBUG
   return 0;
 #elif defined(_WIN32)
-  return timeGetTime();
+  return (unsigned int)GetTickCount64();
 #else
   struct tms t;
   if (!clk_tck) clk_tck = sysconf(_SC_CLK_TCK);
--- /dev/null
+++ b/librtmp/uwpapi.cpp
@@ -0,0 +1,31 @@
+
+#include <windows.h>
+#include <malloc.h>
+
+#if defined(WINAPI_FAMILY) && (WINAPI_FAMILY == WINAPI_FAMILY_APP)
+
+extern "C" {
+  char *get_home_path(void)
+  {
+    auto localFolder = Windows::Storage::ApplicationData::Current->LocalFolder;
+    int len = WideCharToMultiByte(CP_UTF8, 0, localFolder->Path->Data(), -1, NULL, 0, NULL, NULL);
+    char *appdir = (char *)malloc(len);
+    if (appdir) {
+      WideCharToMultiByte(CP_UTF8, 0, localFolder->Path->Data(), -1, appdir, len, NULL, NULL);
+    }
+    return appdir;
+  }
+
+  const char *get_home_drive(void)
+  {
+    auto localFolder = Windows::Storage::ApplicationData::Current->LocalFolder;
+    int len = WideCharToMultiByte(CP_UTF8, 0, localFolder->Path->Data(), 2, NULL, 0, NULL, NULL);
+    char *appdir = (char *)malloc(len);
+    if (appdir) {
+      WideCharToMultiByte(CP_UTF8, 0, localFolder->Path->Data(), 2, appdir, len, NULL, NULL);
+    }
+    return appdir;
+  }
+}
+
+#endif
