From: Alwin Esch <alwin.esch@web.de>
Date: Wed, 8 Jan 2020 12:13:40 +0000
Subject: [PATCH] fix ARM build

---
 context.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/context.h b/context.h
index 33294c1..f959de2 100644
--- a/context.h
+++ b/context.h
@@ -62,7 +62,7 @@
 #endif
 
 #if defined(_ARM_) || defined(ARM) || defined(_M_ARM) || defined(_M_ARM64)
-#define PTW32_PROGCTR(Context)  ((Context).Pc)
+#define  __PTW32_PROGCTR(Context)  ((Context).Pc)
 #endif
 
 #if !defined (__PTW32_PROGCTR)
-- 
2.19.2.windows.1

