--- a/Makefile.am
+++ b/Makefile.am
@@ -22,8 +22,6 @@
 	src \
 	$(proxy_dirs) \
 	contrib \
-	examples \
-	test \
 	po \
 	doc
 
--- a/configure.ac
+++ b/configure.ac
@@ -522,7 +522,6 @@
 AC_CONFIG_FILES([
   Makefile
   contrib/Makefile
-  examples/Makefile
   daemon/Makefile
   daemon/zvbid.init
   doc/Doxyfile
@@ -530,7 +529,6 @@ AC_CONFIG_FILES([
   m4/Makefile
   src/Makefile
   src/dvb/Makefile
-  test/Makefile
   po/Makefile.in
   zvbi.spec
   zvbi-0.2.pc
